### PR TITLE
Use enums for blueprint methods

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine/src/transaction/builder.rs [hunk 1]
```diff
@@ -380,11 +384,22 @@ impl TransactionBuilder {
...
+        for arg in arguments {
+            fields.push(::sbor::decode_any(&arg).unwrap());
+        }
+        let variant = ::sbor::Value::Enum {
+            name: method.to_owned(),
+            fields
+        };
+        let mut bytes = Vec::new();
+        let mut enc = ::sbor::Encoder::with_type(&mut bytes);
+        ::sbor::encode_any(None, &variant, &mut enc);
```
Comment (iamyulong) [inline radix-engine/src/transaction/builder.rs:397]: Minor: duplicate logic

File: scrypto/src/resource/system.rs [hunk 2]
```diff
@@ -48,8 +46,12 @@ impl ResourceSystem {
...
         let input = InvokeSNodeInput {
             snode_ref: SNodeRef::ResourceStatic,
-            function: "main".to_string(),
-            args: args![ResourceManagerFunction::Create(resource_type, metadata, authorization, mint_params)],
+            call_data: scrypto_encode(&ResourceManagerFunction::Create(
+                resource_type,
+                metadata,
+                authorization,
+                mint_params
+            )),
```
Comment (iamyulong) [inline scrypto/src/resource/system.rs:54]: Minor: macro?

File: transaction-manifest/src/generator.rs [hunk 3]
```diff
@@ -282,17 +282,27 @@ pub fn generate_instruction(
...
+                fields.push(validated_arg.dom);
             }
+
+            let variant = ::sbor::Value::Enum {
+                name: generate_string(function)?,
+                fields
+            };
+            let mut bytes = Vec::new();
+            let mut enc = ::sbor::Encoder::with_type(&mut bytes);
+            ::sbor::encode_any(None, &variant, &mut enc);
```
Comment (iamyulong) [inline transaction-manifest/src/generator.rs:300]: I'm now convinced we should have a function for this :)

**Review by iamyulong — COMMENTED** [review]
LGTM

File: transaction-manifest/src/generator.rs (hunk 3 above)
Comment (talekhinezh) [reply transaction-manifest/src/generator.rs:300]: ha yes, okay

**Review by iamyulong — APPROVED** [review]

**Pull request merged**